st2common/st2common/config.py

```diff
@@ -94,6 +94,12 @@ def register_opts(ignore_errors=False):
         cfg.StrOpt('db_name', default='st2', help='name of database'),
         cfg.StrOpt('username', help='username for db login'),
         cfg.StrOpt('password', help='password for db login'),
+        cfg.IntOpt('connection_retry_max_delay_m', help='Connection retry total time (minutes).',
```

Comment:
On a related note - we should update existing code to also use the same `retrying` library in other places (Mistral, querier, etc.). Right now the same logic is implemented a bit differently in at least two other places which is, at the minimum, a maintenance headache.